trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/S3RequestInformation.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.rest;
+
+import com.google.common.base.Splitter;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.UriBuilder;
+import org.glassfish.jersey.server.ContainerRequest;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.trino.s3.proxy.server.collections.MultiMapHelper.lowercase;
+import static java.util.Objects.requireNonNull;
+
+public record S3RequestInformation(String bucketName, String pathInBucket, MultivaluedMap<String, String> requestHeaders, MultivaluedMap<String, String> requestQueryParameters,
```

Comment:
`S3RequestMetadata`?

Reply:
`S3RequestInformation` does not have the body of the request at present, but I think as part of #37 the request body would also be put into it - so it would not be metadata only.

What do you think about `S3Request`?

Reply:
Maybe `ParsedS3Request` or `InternalS3Request`? Or leave it as is is fine too.

Reply:
`ParsedS3Request` sounds good!